src/access/profiling/plotting_utils.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+
+def calculate_column_widths(table_data: list[list], first_col_flexible: bool = True) -> list:
+    """Calculate column widths based on content length
+
+    Args:
+        table_data (list[list]): Table data including headers
+        first_col_flexible (bool): If True, first column gets remaining space after others are sized
+
+    Returns:
+        list : Column width ratios
```

Comment:
Ratios might be better as "fractions". And a note saying that the sum of all the column widths will be 1.0